cypress/GenerateCypressTestSpecs.js

```diff
@@ -0,0 +1,121 @@
+// It is looking for lines in the files that look like this:
+//  Cypress.TestCase(testGroupName, testDescription, testFunction)
+//
+// It uses that data to create:
+//   1) The group subfolders within "cypress/integration".
+//   2) An individual test spec file for each, in that same folder as 
+//      expected by Cypress.
+//   3) The test list: masterListOfAllTestCases
+// ----------------------------------------------------------------------
+
+var fs = require('fs')
+
+const pathToTestFiles = './cypress/tests'
+var fullTestList = []
+
+main();
+WatchAndRun('./cypress/tests', main)
+
+// ----------------------------------------------------------------------
+
+function WatchAndRun(pathToWatch, functionToRun) {
+  var timeoutHandle;
+  var timeOfLastChange;
+  
+  fs.watch(pathToWatch, (eventType, filename) => {
+    var now = new Date();
+    if (timeOfLastChange && now - timeOfLastChange < 1000) { clearTimeout(timeoutHandle); }
+    timeOfLastChange = now;
+    timeoutHandle = setTimeout(functionToRun, 1000)
+  });
+}
+
+// ----------------------------------------------------------------------
+
+function main() {
+  try {
+    // Get the test list file that we MIGHT need to modify.
+    const pathToTestList = './cypress/TestLists.js';
+    var testList_jsContents = fs.readFileSync(pathToTestList, 'ascii');
+    // console.log(testList_jsContents)
+    // console.log()
+
+    // Get the list of test js files.
+    var testFileDirectory = fs.readdirSync(pathToTestFiles);
+    // console.log(testFileDirectory);
+    // console.log()
+
+    // Parse each test js file.
+    testFileDirectory.forEach(fileName => { ParseTestJsFile(fileName); });
+
+    // Finalize our findings by re-writing out the TestList.js file ONLY if it has changed.
+    var index = testList_jsContents.indexOf('// ************ Generated Code Beyond this Point *************************');
+    var newFileContents =
+      '// ************ Generated Code Beyond this Point *************************\r\n' +
+      '// Do NOT manually alter this file from this point onwards.\r\n' +
+      '// Any changes you make will be overridden at runtime.\r\n' +
+      'export const masterListOfAllTestCases = [\r\n';
+    fullTestList.forEach(testSpecification => { newFileContents += `  '${testSpecification}',\r\n` });
+    newFileContents += '];\r\n';
+
+    // Only write the file out if something has changed.
+    if (!testList_jsContents.endsWith(newFileContents)) {
+      newFileContents = testList_jsContents.substring(0, index) + newFileContents;
+      console.log(`TestList.js has been re-generated`)
+      fs.writeFileSync(pathToTestList, newFileContents)
+    }
+    else console.log('NOTHING HAS CHANGED')
+  }
+  catch(error) { console.log(`Caught an Exception:\r\n${error}`)}
+}
+
+// ----------------------------------------------------------------------
+
+function ParseTestJsFile(fileName) {
+  var fileContents = fs.readFileSync(`${pathToTestFiles}/${fileName}`, 'ascii')
```

Comment:
Curios why we have to load as `ascii`? I would think `utf8` would work.

Reply:
Isn't ascii the encoding vsCode is writing files out as?
I switched it to utf8 and tried it, but whenever I edit the file this code is using I see a message (generated by my test code) that the file had changed in the area of the file that I did not change, so I'll keep ascii unless there is a compelling reason to go to utf8.